packages/node_modules/@node-red/runtime/lib/storage/localfilesystem/util.js

```diff
@@ -79,25 +80,31 @@ module.exports = {
      * the write hits disk.
      */
      writeFile: function(path,content,backupPath) {
-         if (backupPath) {
+        if (backupPath) {
             if (fs.existsSync(path)) {
                 fs.renameSync(path,backupPath);
             }
         }
         return when.promise(function(resolve,reject) {
-            var stream = fs.createWriteStream(path);
-            stream.on('open',function(fd) {
-                stream.write(content,'utf8',function() {
-                    fs.fsync(fd,function(err) {
-                        if (err) {
-                            log.warn(log._("storage.localfilesystem.fsync-fail",{path: path, message: err.toString()}));
-                        }
-                        stream.end(resolve);
+            fs.ensureDir(fspath.dirname(path), (err)=>{
```

Comment:
What is the scenario that requires this check? We've never needed it and not had, to my knowledge, any questions about needing to create folders before starting Node-RED.

Reply:
ok, I THINK this was triggered by specifying a specific flow file in settings.js in a place which did not yet exist.
It would be much better done ONCE for any configured things at startup rather than every time a file is written, but then I thought ensureFile is not expensive if the folder does exist, so why not belt-and-braces for writes in localstorage?

What it did do was (I think) blow up rather than be graceful about it.  Maybe excepted in fs.createWriteStream rather than call error?